### ISSUE TITLE
Asynchronous meta reducer support

### ISSUE THREAD

**Issue by danday74** [ngrx/platform#1649]
Meta reducers seemingly don't support async operations. Therefore, I have to create multiple actions to overcome this limitation. You should be able to perform an async operation within a meta reducer that will allow, for example, the payload to be adjusted based on the response of an HTTP request. Such an approach would remove the boilerplate required to have multiple actions and effects to handle this. Please see https://stackoverflow.com/questions/55319705/ngrx-asynchronous-meta-reducer for an example of the problem I am facing.

## Describe any alternatives/workarounds you're currently using

The workaround I am using is described in the Stack Overflow issue.

## Other information:

Redux supports using promises as action payload's and resolving those promises within middleware before passing the result to the reducers. See libs such as redux-promise and redux-thunk - Can ngrx please support this? It will remove the need for a lot of boilerplate.

## If accepted, I would be willing to submit a PR for this feature

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[x ] No

I don't think I have the technical know how to implement this.

**Comment by danday74** [comment]
Why has this been closed? This is a feature request.

In reference to you comment ... "As the answer stated" ... this is a workaround and not an answer, you will notice that the answer was provided by me to my own question.

There are benefits of meta reducers being asynchronous. For example, in the scenario given in the Stack Overflow issue, use of an asynchronous meta reducer would remove the boilerplate of having a separate loaded action. Redux and other store based libs support asynchronous operations in meta reducers. In addition, they support promises as payloads which can be resolved within meta reducers. Please reopen this and provide some consideration as to whether this feature can be implemented. Thanks.

**Comment by pjc2007** [comment]
I wanted to use an async meta reducer to save/restore state from Ionic storage, (ie SQLite) and these operations are async.

**Comment by danday74** [comment]
This is the biggest failing of the otherwise superb ngrx lib IMO.

**Comment by danday74** [comment]
@pjc2007 as @brandonroberts pointed out you can use ngrx Effects to achieve this but it adds a lot of boilerplate.

**Comment by pjc2007** [comment]
@danday74  yes I spent most of the day removing the meta reducer once I realised it can't possibly work with async calls to a database.

I put the loading of the `states`  via an `effect` in response to an action to "hydrate". The saving I put in a `select` rather than an `effect`, as I want to save it out after the reducers have run, and the word was not to dispatch actions from the reducers. 

A bit asymmetric, but I think it is going to work..
